sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -3591,7 +3591,8 @@ class Analyzer(override val catalogManager: CatalogManager)
         case u @ UpCast(child, _, _) if !child.resolved => u
 
         case UpCast(_, target, _) if target != DecimalType && !target.isInstanceOf[DataType] =>
-          throw QueryCompilationErrors.unsupportedAbstractDataTypeForUpCastError(target)
+          throw new AnalysisException(
```

Comment:
Reverted changes as it was in https://github.com/apache/spark/pull/28572

Reply:
@MaxGekk can this be user-facing error via https://github.com/apache/spark/blob/660d2ab07071238dadfd243cd763bb7706637518/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala#L890-L898? Seems like some commands and catalog API to get the view.

Reply:
I haven't find a way to trigger the exception from user space yet. Any help is welcome.